### PR TITLE
More control over external data size

### PR DESCRIPTION
**Pull request by dbogunowicz**
As discussed recently:
- let's fold the splitting of big internal files into `onnx_save` 
- split the model.onnx into graph and constant tensors once the model.onnx is larger then DUMP_EXTERNAL_DATA_THRESHOLD (500MB)
- let's allow the user to control the size of created external tensor files 
 For usage see: https://github.com/neuralmagic/sparseml/pull/1983

### CODE REVIEW

**Comment by mgoin** [comment]
@bfineran @markurtz this seems quite small, i would recommend at least a few gigabytes, if not the relatively standard 10GB used in HF

**Comment by rahul-tuli** [comment]
I thought we were splitting the model into data files if it was >400mb and not restricting the model.data file(s) sizes? The only restriction on model.data file sizes is for easier download/upload from/to sparsezoo for which 400mb seems small? Thoughts?

**Review by bfineran — CHANGES_REQUESTED** [review]
noting we shouldn't be updating chunk size here as discussed via slack - @dbogunowicz is aware of changes

**Comment by dbogunowicz** [comment]
@alexm-nm just giving you the heads up, that splitting the external data is now part of the save_onnx function

File: src/sparsezoo/utils/onnx/external_data.py [hunk 1]
```diff
@@ -98,13 +112,15 @@ def save_onnx(
...
         return True
 
-    if model.ByteSize() > onnx.checker.MAXIMUM_PROTOBUF:
+    if model.ByteSize() > DUMP_EXTERNAL_DATA_THRESHOLD:
         external_data_file = external_data_file or EXTERNAL_ONNX_DATA_NAME
         _LOGGER.warning(
             "The ONNX model is too large to be saved as a single protobuf. "
-            f"Saving with external data: {external_data_file}"
+            "Saving with external data, with file chunks of maximum size "
+            f"{DUMP_EXTERNAL_DATA_THRESHOLD / 1e9} GB"
```
Comment (bfineran) [inline src/sparsezoo/utils/onnx/external_data.py:123]: wrong threshold - also let's downgrade from warning - probably a bit to scary for an expected, default flow

File: src/sparsezoo/utils/onnx/external_data.py [hunk 2]
```diff
@@ -98,13 +112,15 @@ def save_onnx(
             all_tensors_to_one_file=True,
             location=external_data_file,
         )
+        split_external_data(model_path, max_file_size=max_external_file_size)
         return True
 
-    if model.ByteSize() > onnx.checker.MAXIMUM_PROTOBUF:
+    if model.ByteSize() > DUMP_EXTERNAL_DATA_THRESHOLD:
         external_data_file = external_data_file or EXTERNAL_ONNX_DATA_NAME
         _LOGGER.warning(
```
Comment (dbogunowicz) [inline src/sparsezoo/utils/onnx/external_data.py:120]:
```suggestion
        _LOGGER.debug(
```

File: src/sparsezoo/utils/onnx/external_data.py (hunk 1 above)
Comment (dbogunowicz) [inline src/sparsezoo/utils/onnx/external_data.py:123]:
```suggestion
            f"{max_external_file_size / 1e9} GB"
```

**Review by bfineran — APPROVED** [review]

**Pull request merged**